lib/FusionInventory/Agent/Task/Inventory/Win32/Networks.pm

```diff
@@ -41,7 +43,15 @@ sub doInventory {
         delete $interface->{DNSDomain};
         delete $interface->{GUID};
 
-        $interface->{TYPE} = _getMediaType($interface->{PNPDEVICEID});
+        # Don't reload registry keys between interfaces checks
+        $keys = getRegistryKey(
+            path   => "HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/Control/Network/{4D36E972-E325-11CE-BFC1-08002BE10318}",
+            wmiopts => { # Only used for remote WMI optimization
+                values  => [ qw/PnpInstanceID MediaSubType/ ]
+            }
+        ) unless $keys;
```

Comment:
That's a bit confusing... Unless the intent is to avoid any registry access in case there is no interface, the registry access should occurs once for all before entering the loop.

Reply:
The `unless` here is a shortcut. I agree this is not nicely readable.

As the requested registry tree won't very probably not change during the interfaces loop (from mediatype check), that's better to only do one access if we find at least one interface.

Actually windows reports a lot of interface, mostly virtual and not activated... This wasn't an issue while getting locally the registry, but that could become really not efficient while doing such in the remote inventory case.